Source/Core/Core/Scripting/CoreScriptInterface/InternalScriptAPIs/ArgHolder_APIs.h

```diff
@@ -0,0 +1,188 @@
+#pragma once
+
+namespace Scripting
+{
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// ArgHolder_APIs contains the APIs for creating/handling an opaque handle to an ArgHolder*
+
+// IMPORTANT NOTE: All functions that start with "Create" allocate a new object on the heap. The
+// DLL is reponsible for freeing them by calling Delete_ArgHolder on them when done using them.
+
+// Also, the void* passed as the 1st parameter to a function represents an opaque handle for
+// an ArgHolder* unless the function's documentation says otherwise.
+typedef struct ArgHolder_APIs
+{
+  // Takes an opaque ArgHolder* as input, and returns its type as an int (which represents an
+  // ArgTypeEnum)
+  int (*GetArgType)(void*);
+
+  // Returns 1 if the ArgHolder* passed into it was empty, and 0 otherwise.
+  int (*GetIsEmpty)(void*);
+
+  // Creates + returns an ArgHolder* which has no value inside of it.
+  void* (*CreateEmptyOptionalArgHolder)();
+
+  // Creates + returns a boolean ArgHolder* with the specified value.
+  void* (*CreateBoolArgHolder)(int);
+
+  // Creates + returns an unsigned_8 ArgHolder* with the specified value.
+  void* (*CreateU8ArgHolder)(unsigned long long);
```

Comment:
maybe just use uint64_t, etc for the relevant functions?

Reply:
For all of the function pointers which are going to be passed across the DLL boundary (such as this one), I tried to make all integer types in the signature/parameters either unsigned char, signed char, unsigned long long, and signed long long. This way, I could be sure that the data I'm passing would be either 1 byte or the maximum bytesize for the system (realistically, 8 bytes for any computer running Dolphin).

Also, I want to make sure that the type exists in plain C, which is why I tried to avoid including other types (ex. I return ints from functions instead of bool, since C doesn't have a bool type).

Reply:
Sorry to necro but one thing is that using variable length types I think causes annoying ABI concerns (especially across languages that may not have ULL as an inbuilt type), While ULL (`unsigned long long`) is guaranteed to be 64 bits or larger, this makes things incompatible if Dolphin is rebuilt and ULL changes.

And realistically if something doesn't have `uint64_t`, it's already a super esoteric implementation of C in the first place.